src/Service.Tests/OpenApiDocumentor/StoredProcedureGeneration.cs

```diff
@@ -0,0 +1,346 @@
+                Assert.AreEqual(expectedParameters.Length, schemaComponentProperties.Count, message: "The number of generated parameters is not expected.");
+
+                // Validate column presence and accurate column JSON type.
+                // Test input expectedColumns and expectedColumnJsonTypes are always expected to have the same length.
+                for (int colIdx = 0; colIdx < expectedParameters.Length; colIdx++)
+                {
+                    string columnName = expectedParameters[colIdx];
+                    string columnType = expectedParametersJsonTypes[colIdx];
+                    Assert.IsTrue(schemaComponentProperties.ContainsKey(columnName), message: "Unexpected parameter absence in request schema component.");
+                    Assert.AreEqual(schemaComponentProperties[columnName].Type, columnType, message: "Unexpected parameter value type in request schema component.");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Validates that the generated response body references stored procedure
+        /// result set columns and not parameters.
+        /// </summary>
+        /// <param name="entityName">Entity to test, requires updating the CreateEntities() helper.</param>
+        /// <param name="expectedColumns">Expected first result set columns</param>
+        /// <param name="expectedColumnJsonTypes">Expected first result set column types (JSON)</param>
+        [DataRow("sp1", new string[] { "outputName" }, new string[] { "string" }, DisplayName = "Validate column and column type in first result set.")]
+        [DataRow("sp2", new string[] { "age" }, new string[] { "number" }, DisplayName = "Validate that a output result set column is resolved separately from an equivalently named parameter.")]
+        [DataTestMethod]
+        public void ValidateResponseBodyContents(string entityName, string[] expectedColumns, string[] expectedColumnJsonTypes)
+        {
+            // With the responses, we can validate the Properties and their types
+            // can also validate Reference.Id 'sp1_sp_reponse' to ensure correct mapping.
+            // Though, the fact that Properties is populated correctly, means the OpenApi.Net mechanisms
+            // populated Properties with the contents of the referenced schema component.
+            OpenApiResponses responses = GetOperationResponses(entityName, OperationType.Get);
+
+            // Validate the correct schema component is referenced in the response body.
+            // The OpenApiResponses dictionary key represents the integer value of the HttpStatusCode,
+            // which is returned when using Enum.ToString("D").
+            // The "D" format specified "displays the enumeration entry as an integer value in the shortest representation possible."
+            OpenApiReference reference = GetResponseBodyReference(responses[HttpStatusCode.OK.ToString("D")]);
```

Comment:
Once the OpenApiReference is obtained, looks like both the previous and this test perform same kind of assertions. Can we move the assertions into a helper?

Reply:
Moved into a common helper. Renamed error messages and helper variables from column/parameters to "Properties" to reflect the changes. No functional changes though. Result is reduced duplicate code, slightly more generic error messages. There are distinct tests for validating request body versus response body so any failures in either will directly correlate to whether a validation issue occurred for parameters versus columns.